### DISCUSSION TITLE
Machine Learning Systems - 1  Introduction

### DISCUSSION THREAD

**Discussion by giscus** [harvard-edge/cs249r_book#292] [Chapter Discussions]
# Machine Learning Systems - 1  Introduction

https://harvard-edge.github.io/cs249r_book/contents/introduction/introduction.html

**Comment by emmanuel2406** [comment]
Am excited to dive into a comprehensive overview of this rich vein within CS ... I can see so much potential for impact on the world!

**Reply by profvjreddi** [reply]
Hi! 👋 I'm thrilled that you're excited to explore the book. Happy reading, and I hope you find it both informative and inspiring! Keep me posted.

**Comment by PandaKungF** [comment]
I am in the very beginning of my ML implementation journey. I am very excited to read your book. Thank you for your work and sharing it!

**Reply by profvjreddi** [reply]
Welcome! I'm excited to have you join us. ML is a fast-paced but highly rewarding subject to study. In this book, we've tried to get down to the key concepts; it is a work in progress though since the field is still evolving. If you have any questions or just want to share your thoughts as you go along, don't hesitate to reach out. Your feedback also means a lot and can help make the book even better. So keep us posted. Feel free to post any thoughts or questions in the discussions section, which's here: https://github.com/harvard-edge/cs249r_book/discussions

Happy reading and enjoy the learning process!

**Comment by Mjrovai** [comment]
This is a Fantastic Introduction to the book. ML systems are definitely the bones of AI. Congratulations, and thanks for such amazing thoughts.

**Comment by SYLee1996** [comment]
Thanks for sharing, this seems like a great learning resource!

**Reply by profvjreddi** [reply]
Thank you, @SYLee1996! It’s a work in progress, and I’m improving it. Are there any specific areas or topics you feel could benefit from deeper exploration? If anything comes to mind, please let me know—I truly value constructive feedback, and it’s what keeps me motivated!

**Comment by GreggatElle** [comment]
Excellent work coming out of the man cave VJ :-)

**Reply by profvjreddi** [reply]
Haha, sometimes you’ve just gotta retreat to the cave to get things done!

**Comment by Imran-Iqbal** [comment]
Awesome! Fantastic! Terrific!

**Reply by profvjreddi** [reply]
Thank you for your enthusiasm and support. They motivate me to keep working on it! 🙏

**Comment by Nandalib** [comment]
I am thrilled to explore your book. Thanks for the comprehensive overview focusing on both theoretical as well as practical  domains. Referring to this information ''This interconnectedness can create either virtuous or vicious cycles. In a virtuous cycle, high-quality data enables effective learning,.....
'' Is that really possible that we could be above to improve the accuracy after quantization? In my case, I have achieved  SMAPE of 7%, marking a 3% improvement compared to the original unoptimized model.

**Reply by profvjreddi** [reply]
Thank you for the feedback! I always appreciate knowing that people are reading and learning from my work because that’s what motivates me. 

> '' Is that really possible that we could be above to improve the accuracy after quantization?

To answer your question, quantization introduces small noise or perturbations to the weights and activations due to reduced precision. This makes it act as a form of regularization, which helps prevent overfitting, particularly in over-parameterized models, which is often the case with many models. The models that overfit the training data often struggle to generalize well to unseen data, so regularization helps cope with this issue. I hope that helps!

**Reply by Nandalib** [reply]
Thank you so much for your feedback!  and I appreciate the opportunity to connect and share ideas.

**Comment by gwq-nb** [comment]
It disappeared

**Reply by profvjreddi** [reply]
What did :)

**Comment by PursuitOfCodeKnowledge** [comment]
What was it? :))

**Comment by TrilhadoGPT** [comment]
Hello everyone! 😊  

I built this GPT using the material from the repository to help with studying.  

Here’s the link:

https://chatgpt.com/g/g-695f93bdfa948191863921ffd00a95da-harvard-ml-coach
